src/libraries/System.Net.Http/src/System/Net/Http/HttpClientHandler.cs

```diff
@@ -26,30 +26,34 @@ public partial class HttpClientHandler : HttpMessageHandler
 
 #if TARGET_BROWSER
         private IMeterFactory? _meterFactory;
-        private MetricsHandler? _metricsHandler;
+        private HttpMessageHandler? _firstHandler; // DiagnosticsHandler or MetricsHandler, depending on global configuration.
 
-        private MetricsHandler Handler
+        private HttpMessageHandler Handler
         {
             get
             {
-                if (_metricsHandler != null)
+                if (_firstHandler != null)
                 {
-                    return _metricsHandler;
+                    return _firstHandler;
                 }
 
                 HttpMessageHandler handler = _underlyingHandler;
+
+                // MetricsHandler should be descendant of DiagnosticsHandler in the handler chain to make sure the 'http.request.duration'
+                // metric is recorded before stopping the request Activity. This is needed to make sure that our telemetry supports Exemplars.
+                handler = new MetricsHandler(handler, _meterFactory, out _);
                 if (DiagnosticsHandler.IsGloballyEnabled())
                 {
                     handler = new DiagnosticsHandler(handler, DistributedContextPropagator.Current);
                 }
-                MetricsHandler metricsHandler = new MetricsHandler(handler, _meterFactory, out _);
 
                 // Ensure a single handler is used for all requests.
-                if (Interlocked.CompareExchange(ref _metricsHandler, metricsHandler, null) != null)
+                if (Interlocked.CompareExchange(ref _firstHandler, handler, null) != null)
                 {
-                    metricsHandler.Dispose();
+                    handler.Dispose();
                 }
-                return _metricsHandler;
+
+                return _firstHandler;
```

Comment:
A conversation for another PR, but why is all of this code here rather than inside of BrowserHttpHandler, just as it's inside of SocketsHttpHandler rather than being in the #else block below?

Reply:
Made a tracking issue: https://github.com/dotnet/runtime/issues/104468